### PR TITLE
enable duplicate packages warning

### PR DESCRIPTION
**Pull request by mkholjuraev**
This PR is intended to add a console warning of the packages loaded multiple times into shared scope. Now the warning is cleaner, and easy to spot, and also includes a command for insights-interact-tool that will make aligning the versions easier. Due to a problem with not being able to access all package versions as keys of shared scope, this feature will be 'opt in' for micro-apps. They need to enable this by explicitly calling the **enablePackagesDebug** function in the chrome API.

image

### CODE REVIEW

File: src/chrome/warnDuplicatePackages.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    );
+
+    const apps = Object.entries(updateCommands);
+
+    apps.forEach(([appName, packages]) => {
+      console.group(
+        `[SCALPRUM]: To align ${appName} application's package dependency versions to the chroming app instance version, please run following command`
+      );
+      console.warn(
+        `%c insights-interact run alignPackages --app=${appName} --packages=${packages} `,
```
Comment (Hyperkid123) [inline src/chrome/warnDuplicatePackages.ts:73]: This is for a future update. Where is the `insights-interact` command available from? Can we add it to the FEC binary so every repository have access to this command?

**Comment by Hyperkid123** [comment]
We are also missing the 
```
enablePackagesDebug: () => warnDuplicatePkg(),
```

In The chrome API typings. You can add it in FEC repo and then bump the types package here.

File: src/chrome/warnDuplicatePackages.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+export const warnDuplicatePkg = () => {
+  const packages: Package = getWebpackScope();
+  const entries = Object.entries(packages);
+  const updateCommands: ToAlign = prepareUpdateCommands(entries);
+  const pkgNames = Object.keys(updateCommands);
+
+  if (pkgNames.length) {
+    console.group(
+      `%c [SCALPRUM]: You have following packages that is being loaded into browser multiple times. You might want to align your version with the chrome one`,
+      'font-size:20px; color:#581845; padding:1px; border-radius:1px;'
```
[the remark this answers is not in the document]
Comment (mkholjuraev) [reply src/chrome/warnDuplicatePackages.ts:63]: No worries, I am open to any improvements :). I thought it is better to give more context for unfamiliar devs, even if the message is long. Can you please suggest a shorter version?

File: src/chrome/warnDuplicatePackages.ts (hunk 1 above)
Comment (mkholjuraev) [reply src/chrome/warnDuplicatePackages.ts:73]: Here is: https://github.com/RedHatInsights/insights-interact-tools/pull/7. This tool can be modified to be used platform wide.

File: src/chrome/warnDuplicatePackages.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hyperkid123) [reply src/chrome/warnDuplicatePackages.ts:63]: The message is OK. The text is just way too big. If we can have it exactly half way between the default size and the 20px  that would be adequate I think.

**Comment by mkholjuraev** [comment]
> We are also missing the
> 
> ```
> enablePackagesDebug: () => warnDuplicatePkg(),
> ```
> 
> In The chrome API typings. You can add it in FEC repo and then bump the types package here.

Here is the PR to add the type: https://github.com/RedHatInsights/frontend-components/pull/1736

**Comment by mkholjuraev** [comment]
@Hyperkid123 the types package is updated and the tests are passing. Please have a look one more time, if this gets merged I am going to demo it tomorrow :)

**Comment by mkholjuraev** [comment]
@Hyperkid123 one last step to merge this, please. I do not have power to do so

**Comment by Hyperkid123** [comment]
@mkholjuraev I hope its not too late for the demo. Sorry I was occupied with the work.

**Pull request merged**